### PR TITLE
Demo App restructured

### PR DESCRIPTION
**Pull request by bielikb**
### 🎯 Goal

Restructure Demo App.

### 📝 Summary

* StreamChat API, components, extensions were consolidated under `StreamChat` folder in `DemoApp`
* `StreamChatWrapper` now represents the integration point for `StreamChat` SDK
* UI screens were separated into single swift files and moved under `Screens` folder
* `DemoAppConfiguration` acts as a single place to configure/initialize configuration used by the iOS dev team == `AppDelegate` became thin.
* `DemoAppCoordinator` now holds onto `StreamChatWrapper` instance and `PushNotifications` instance. Several responsibilities handled by coordinator were moved onto respective interfaces (wrapper or push notifications) == better separation of concerns.

### 🎨 Showcase

| Before | After |
| ------ | ----- |
| Screenshot 2022-06-08 at 16 48 01  |  Screenshot 2022-06-08 at 16 47 16 |

### 🧪 Manual Testing Notes

_Explain how this change can be tested manually, if applicable._

### ☑️ Contributor Checklist
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] This change follows zero ⚠️ policy (required)
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_

### CODE REVIEW

File: DemoApp/DemoAppConfiguration.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            }
+        }
+
+        // Performance tracker
+        func trackPerformanceIfNeeded() {
+            if isStreamInternalConfiguration {
+                PerformanceMonitor.shared().performanceViewConfigurator.options = [.performance]
+                PerformanceMonitor.shared().start()
+            }
+        }
```
Comment (evsaev) [inline DemoApp/DemoAppConfiguration.swift:56]: We declare several func level funcs but we don't call them anywhere 🤔

File: DemoApp/PushNotifications.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        UNUserNotificationCenter.current()
+    }
+
+    var authorizationOptions: UNAuthorizationOptions {
+        [.alert, .sound, .badge]
+    }
+
+    var onNotificationResponse: ((UNNotificationResponse) -> Void)?
+
+    static let shared = PushNotifications()
```
Comment (evsaev) [inline DemoApp/PushNotifications.swift:20]: I would be avoiding singletons. Why not to have it as a property in `AppDelegate`?

File: DemoApp/DemoAppCoordinator.swift [hunk 3]
```diff
@@ -34,33 +32,22 @@ final class DemoAppCoordinator: NSObject {
...
+    func handlePushNotificationResponse() {
+        pushNotifications.onNotificationResponse = { [weak self] response in
+            guard case UNNotificationDefaultActionIdentifier = response.actionIdentifier else {
+                return
+            }
 
-        guard let cid = notificationInfo.cid else {
-            return
-        }
+            guard let self = self else { return }
```
Comment (evsaev) [inline DemoApp/DemoAppCoordinator.swift:42]: I would do that as the first thing in closure and move checking `actionIdentifier` to the next guard OR event have a single guard for all statements

File: DemoApp/StreamChat/Extensions/DeletedMessageVisibility+CustomStringConvertible.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+
+extension ChatClientConfig.DeletedMessageVisibility: CustomStringConvertible {
```
Comment (evsaev) [inline DemoApp/StreamChat/Extensions/DeletedMessageVisibility+CustomStringConvertible.swift:8]: What if we just make the change in LLC? When this setting is moved to the backend, string would be the way to encode/decode it from JSON so it's fine to make this change, I think
```swift
public enum DeletedMessageVisibility: String {
        /// All deleted messages are always hidden.
        case alwaysHidden
        /// Deleted message by current user are visible, other deleted messages are hidden.
        case visibleForCurrentUser
        /// Deleted messages are always visible.
        case alwaysVisible
    }
```

File: DemoApp/StreamChat/StreamChatWrapper.swift [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        client = nil
+    }
+}
+
+// MARK: Controllers
+
+extension StreamChatWrapper {
+    func channelController(for channelId: ChannelId?) -> ChatChannelController? {
+        channelId.map { client.channelController(for: $0) }
+    }
```
Comment (evsaev) [inline DemoApp/StreamChat/StreamChatWrapper.swift:122]: I would make `channelId` non-optional and handle `nils` on the caller side. Asking the wrapper to give a controller for `nil` is a bit weird

File: DemoApp/StreamChat/StreamChatWrapper.swift [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    func channelListController(query: ChannelListQuery) -> ChatChannelListController {
+        client.channelListController(query: query)
+    }
+}
+
+// MARK: Push Notifications
+
+extension StreamChatWrapper {
+    func setupRemoteNotifications() {
+        PushNotifications.shared.registerForPushNotifications()
```
Comment (evsaev) [inline DemoApp/StreamChat/StreamChatWrapper.swift:133]: what if we pass `PushNotifications` instance as a dependency via `init`?

**Review by evsaev — APPROVED** [review]
Looks good 💪 Left some comments, mostly minor

File: DemoApp/StreamChat/Extensions/DeletedMessageVisibility+CustomStringConvertible.swift (hunk 4 above)
Comment (bielikb) [reply DemoApp/StreamChat/Extensions/DeletedMessageVisibility+CustomStringConvertible.swift:8]: Im happy to do so, if there are no objections from the team.

Comment (bielikb) [reply DemoApp/StreamChat/Extensions/DeletedMessageVisibility+CustomStringConvertible.swift:8]: This small bit in our demo app pinpoints another deficiency in our LLC layer. 
Our models aren't `CustomDebugStringConvertible`, thus they can't be nicely interpreted by the LLDB.

File: DemoApp/DemoAppConfiguration.swift (hunk 1 above)
Comment (bielikb) [reply DemoApp/DemoAppConfiguration.swift]: Thanks! This must've slipped through my fingers :) Done!

File: DemoApp/DemoAppCoordinator.swift (hunk 3 above)
Comment (bielikb) [reply DemoApp/DemoAppCoordinator.swift]: Done!

File: DemoApp/StreamChat/Extensions/DeletedMessageVisibility+CustomStringConvertible.swift (hunk 4 above)
Comment (bielikb) [reply DemoApp/StreamChat/Extensions/DeletedMessageVisibility+CustomStringConvertible.swift]: I moved that conformance to LLC

File: DemoApp/StreamChat/StreamChatWrapper.swift (hunk 6 above)
Comment (bielikb) [reply DemoApp/StreamChat/StreamChatWrapper.swift]: Thanks. I provided closure to fire this logic to `StreamChatWrapper` to which the outer clients can hook into.

File: DemoApp/PushNotifications.swift (hunk 2 above)
Comment (bielikb) [reply DemoApp/PushNotifications.swift]: Done ;)

File: DemoApp/StreamChat/StreamChatWrapper.swift (hunk 5 above)
Comment (bielikb) [reply DemoApp/StreamChat/StreamChatWrapper.swift:125]: Good catch! Done ;)

**Pull request merged**